doc/spec_TransferElement.dox

```diff
@@ -385,6 +386,12 @@ This document describes the behaviour of the TransferElement base class, the NDR
     - 5.2.7 A write operation is executed on a transfer element that cannot be written
       - 5.2.7.1 Check with TransferElement::isWriteable() or TransferElement::isReadOnly()
       - \anchor transferElement_C_5_2_7_2 5.2.7.2 Thrown in doPreWrite() [\ref UnifiedTest_TransferElement_C_5_2_7_2 "U"]
+    - 5.2.8 Scalar, 1D or 2D registers of UserType Void
```

Comment:
- Try not to use the enumeration as sub-header. Directly write the testable sentence.
- Be very exact: It's not Scalar register, but ScalarRegisterAccessor

This section is about when a runtime error is thrown. Each point follows "They are thrown if and only if one of the following conditions are met:" and must be phrased like that.

- 5.2.8 ScalarRegisterAccessor, OneDRegisterAccessor of TwoDRegisterAccessor are instantiated with UserType ChimeraTK::Void. *footnote* (Reason: The data accessing members conceptually don't make sense. Some of them like operator++ etc are not even imeplemtable. VoidRegisterAccessor must me used instead)
   - Throws in the constructor

Reply:
Is this really about runtime errors? Since template instantiation with UserType=Void happens at compile time, you could already throw an error then if Scalar/1D/2D RegisterAccessor are instantiated. So in the spec, it would simply read "instantiation of Scalar/1D/2D RegisterAccessor with UserType=Void is forbidden". Not referring to any runtime error.

Reply:
Your are right. This section is explicitly NOT about runtime errors but about LOGIC errors. Sorry for the confusion.

Reply:
> Is this really about runtime errors? Since template instantiation with UserType=Void happens at compile time, you could already throw an error then if Scalar/1D/2D RegisterAccessor are instantiated. So in the spec, it would simply read "instantiation of Scalar/1D/2D RegisterAccessor with UserType=Void is forbidden". Not referring to any runtime error.

I like this short version, although I would add a footnote. The Dox file is different than what I expected, got to google the structure of footnootes.